core/src/main/java/io/debezium/connector/cassandra/CassandraSchemaFactory.java

```diff
@@ -18,4 +49,380 @@ public CassandraSchemaFactory() {
+            this.startRange = startRange;
+        }
+
+        public void addEndRange(RangeData endRange) {
+            this.endRange = endRange;
+        }
+
+        public RangeData getStartRange() {
+            return startRange;
+        }
+
+        public RangeData getEndRange() {
+            return endRange;
+        }
+
+        public void addCell(CellData cellData) {
+            this.cellMap.put(cellData.name, cellData);
+        }
+
+        public void removeCell(String columnName) {
+            if (hasCell(columnName)) {
+                cellMap.remove(columnName);
+            }
+        }
+
+        public boolean hasCell(String columnName) {
+            return cellMap.containsKey(columnName);
+        }
+
+        public boolean hasAnyCell() {
+            return !cellMap.isEmpty();
+        }
+
+        @Override
+        public Struct record(Schema schema) {
+            Struct struct = new Struct(schema);
+            for (Field field : schema.fields()) {
+                Schema cellSchema = KeyValueSchema.getFieldSchema(field.name(), schema);
+                if (field.name().equals(RANGE_START_NAME) && startRange != null) {
+                    struct.put(field.name(), startRange.record(cellSchema));
+                }
+                else if (field.name().equals(RANGE_END_NAME) && endRange != null) {
+                    struct.put(field.name(), endRange.record(cellSchema));
+                }
+                else {
+                    CellData cellData = cellMap.get(field.name());
+                    if (cellData != null) {
+                        struct.put(field.name(), cellData.record(cellSchema));
+                    }
+                }
+            }
+            return struct;
+        }
+
+        public RowData copy() {
+            RowData copy = new RowData();
+            for (CellData cellData : cellMap.values()) {
+                copy.addCell(cellData);
+            }
+            return copy;
+        }
+
+        /**
+         * Assemble the Kafka connect {@link Schema} for the "after" field of the change event
+         * based on the Cassandra table schema.
+         *
+         * @param tm metadata of a table that contains the Cassandra table schema
+         * @return a schema for the "after" field of a change event
+         */
+        static Schema rowSchema(TableMetadata tm) {
+            List<String> columnNames = new ArrayList<>();
+            List<DataType> columnTypes = new ArrayList<>();
+
+            for (ColumnMetadata cm : tm.getColumns().values()) {
+                columnNames.add(cm.getName().toString());
+                columnTypes.add(cm.getType());
+            }
+
+            return rowSchema(columnNames, columnTypes);
+        }
+
+        static Schema rowSchema(List<String> columnNames, List<DataType> columnsTypes) {
+            SchemaBuilder schemaBuilder = SchemaBuilder.struct().name(Record.AFTER).version(1);
```

Comment:
@smiklosovic I am sorry for the late reaction. There is a small thing to refactor if possible. Could you please extract the version numbers into constants in the same way as is done in `SchemaFactory` class? So in general the constant definitions will always be schema name followed by schema version separated by en empty line and ideally with a comment added if there are logical groups of schemas.
This would help in future with schema management and maintenance. The code functionality is otherwise good.
Thanks for understaning.

Reply:
@jpechane yes I will take a look.